Source/Events.Store.MongoDB/Processing/Streams/StreamProcessorStates.cs

```
@@ -27,60 +29,125 @@ public class StreamProcessorStates : IStreamProcessorStateRepository
-                // var state = await states.Find(
-                //     _streamProcessorFilter.Eq(_ => _.ScopeId, streamProcessorId.ScopeId.Value)
-                //         & _streamProcessorFilter.Eq(_ => _.EventProcessorId, streamProcessorId.EventProcessorId.Value)
-                //         & _streamProcessorFilter.Eq(_ => _.SourceStreamId, streamProcessorId.SourceStreamId.Value))
-                //     .FirstOrDefaultAsync(cancellationToken)
-                //     .ConfigureAwait(false);
+                if (id is SubscriptionId subscriptionId)
+                {
+                    if (baseStreamProcessorState is StreamProcessorState streamProcessorState)
+                    {
+                        var replacementState = new SubscriptionState(
+                            subscriptionId.ConsumerTenantId,
+                            subscriptionId.ProducerMicroserviceId,
+                            subscriptionId.ProducerTenantId,
+                            subscriptionId.ScopeId,
+                            subscriptionId.StreamId,
+                            subscriptionId.PartitionId,
+                            streamProcessorState.Position,
+                            streamProcessorState.RetryTime,
+                            streamProcessorState.FailureReason,
+                            streamProcessorState.ProcessingAttempts);
+                        var states = await _connection.GetSubscriptionStateCollection(replacementState.ScopeId, cancellationToken).ConfigureAwait(false);
+                        var persistedState = await states.ReplaceOneAsync(
+                            CreateFilter(subscriptionId),
+                            replacementState,
+                            new ReplaceOptions { IsUpsert = true })
+                            .ConfigureAwait(false);
+                    }
```

It should throw an exception if stream processor state is not StreamProcessorState